configure.ac

```diff
@@ -305,6 +306,42 @@ AC_CHECK_HEADER(valgrind/memcheck.h, [],
     AC_MSG_ERROR([valgrind requested but <valgrind/memcheck.h> not found.]))
 fi
 
+dnl Check for userfaultfd support
+have_userfaultfd=0
+AC_CHECK_HEADERS([linux/userfaultfd.h],
+    [AC_MSG_CHECKING([for userfaultfd interface])
+     AC_TRY_LINK([#include <sys/types.h>
+#include <linux/userfaultfd.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <fcntl.h>
+],
+        [return syscall(__NR_userfaultfd, O_CLOEXEC | O_NONBLOCK);],
```

Comment:
FWIW, `AC_TRY_LINK` is deprecated -- `AC_LINK_IFELSE` is preferred these days.

That being said, do you need to try to link?  You're really just checking for all those header files and `__NR_userfaultfd`, right?  Because `syscall()` will definitely exists, as well `O_CLOEXEC` and `O_NONBLOCK`, right?  I.e., is there a reason not to use `AC_COMPILE_IFELSE`?  Or even just `AC_CHECK_HEADERS` and `AC_CHECK_DECL` (for `__NR_userfaultfd`)?

Perhaps something like (this is off the top of my head / not tested / YMMV):

```sh
have_userfault_fd=0
have_userfault_fd_msg=no
AC_MSG_CHECKING([for userfaultfd interface])
AC_CHECK_HEADER([linux/userfaultfd.h],
    [AC_CHECK_DECL([__NR_userfaultfd], [have_userfault_fd=1 have_userfault_fd_msg=yes])])
AC_MSG_RESULT([$have_userfault_fd_msg])
```

(same comments apply to your use of `AC_TRY_LINK` below)

Reply:
Yeah, ok, some of us are old and still use TRY_LINK, but good call out.

I think AC_CHECK_DECL is probably ok, I can't remember why I wanted to link instead of check the declaration.